libraries/AP_Math/control.cpp

```diff
@@ -22,18 +22,203 @@
 #include "AP_Math.h"
 #include "vector2.h"
 #include "vector3.h"
+#include <AP_InternalError/AP_InternalError.h>
 
 // control default definitions
-#define CONTROL_TIME_CONSTANT_RATIO 4.0f   // minimum horizontal acceleration in cm/s/s - used for sanity checking acceleration in leash length calculation
+#define CONTROL_TIME_CONSTANT_RATIO 4.0f   // time constant to ensure stable kinimatic path generation
 
 // update_pos_vel_accel - single axis projection of position and velocity, pos and vel, forwards in time based on a time step of dt and acceleration of accel.
-void update_pos_vel_accel(float& pos, float& vel, float accel, float dt)
+void update_vel_accel(float& vel, float accel, float dt,
+    bool limit_neg, bool limit_pos, float vel_error)
+{
+    float delta_vel = accel * dt;
```

Comment:
nitpick: we can put a const before, "float delta_vel"

Reply:
done